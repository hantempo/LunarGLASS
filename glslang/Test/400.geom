#version 400 core

void main()
{
    EmitStreamVertex(1);
    EndStreamPrimitive(0);
    EmitVertex();
    EndPrimitive();

    int id = gl_InvocationID;
}

layout(invocations = 3) out outbn { int a; }; // ERROR, not on a block
layout(max_vertices = 127, invocations = 4) out;

#extension GL_ARB_separate_shader_objects : enable

in gl_PerVertex {      // testing input arrays with a block redeclaration, see 420.geom for without
    vec4 gl_Position;
} gl_in[];

void foo()
{
    gl_in.length();  // ERROR
    gl_in[1].gl_Position;
}

in vec4 color[];
in vec4 color2[];
in vec4 colorS[3];
in vec4 colorBad[4];

void foo2()
{
    color.length(); // ERROR
    colorS.length();
}

layout(triangles) in;  // give ERROR just for colorBad

in vec4 color[3];
in vec4 color2[3];
in vec4 colorbad2[2];  // ERROR

void foo3()
{
    gl_in.length();
    color.length();
    color2.length();
    colorS.length();
}
