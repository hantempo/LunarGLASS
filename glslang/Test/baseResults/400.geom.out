400.geom
Warning, version 400 is not yet complete; some version-specific features are present, but many are missing.
ERROR: 0:13: 'invocations' : can only apply to a standalone qualifier 
ERROR: 0:24: 'length' :  array must be declared with a size before using this method
ERROR: 0:35: 'length' :  array must be declared with a size before using this method
ERROR: 0:39: 'triangles' : inconsistent input primitive for array size colorBad
ERROR: 0:43: 'triangles' : inconsistent input primitive for array size colorbad2
ERROR: 5 compilation errors.  No code generated.


invocations = 4
max_vertices = 127
input primitive = triangles
output primitive = none
ERROR: node is still EOpNull!
0:3  Function Definition: main( (void)
0:3    Function Parameters: 
0:5    Sequence
0:5      EmitStreamVertex (void)
0:5        Constant:
0:5          1 (const int)
0:6      EndStreamPrimitive (void)
0:6        Constant:
0:6          0 (const int)
0:7      EmitVertex (void)
0:8      EndPrimitive (void)
0:10      Sequence
0:10        move second child to first child (int)
0:10          'id' (int)
0:10          'gl_InvocationID' (in int)
0:22  Function Definition: foo( (void)
0:22    Function Parameters: 
0:24    Sequence
0:24      Constant:
0:24        1 (const int)
0:25      gl_Position: direct index for structure (4-component vector of float)
0:25        direct index (block{gl_Position})
0:25          'gl_in' (in 3-element array of block{gl_Position})
0:25          Constant:
0:25            1 (const int)
0:25        Constant:
0:25          0 (const int)
0:33  Function Definition: foo2( (void)
0:33    Function Parameters: 
0:35    Sequence
0:35      Constant:
0:35        1 (const int)
0:36      Constant:
0:36        3 (const int)
0:45  Function Definition: foo3( (void)
0:45    Function Parameters: 
0:47    Sequence
0:47      Constant:
0:47        3 (const int)
0:48      Constant:
0:48        3 (const int)
0:49      Constant:
0:49        3 (const int)
0:50      Constant:
0:50        3 (const int)
0:?   Linker Objects
0:?     '__anon__0' (layout(stream=0 ) out block{a})
0:?     'gl_in' (in 3-element array of block{gl_Position})
0:?     'color' (in 3-element array of 4-component vector of float)
0:?     'color2' (in 3-element array of 4-component vector of float)
0:?     'colorS' (in 3-element array of 4-component vector of float)
0:?     'colorBad' (in 4-element array of 4-component vector of float)
0:?     'colorbad2' (in 2-element array of 4-component vector of float)


Linked geometry stage:

ERROR: Linking geometry stage: At least one geometry shader must specify an output layout primitive


